app/grandchallenge/algorithms/views.py

```diff
@@ -478,28 +479,84 @@ def get_success_url(self):
         return self.algorithm.get_absolute_url()
 
 
+class JobCreatePermissionMixin(
+    LoginRequiredMixin, VerificationRequiredMixin, AccessMixin
+):
+    @cached_property
+    def algorithm(self) -> Algorithm:
+        return get_object_or_404(Algorithm, slug=self.kwargs["slug"])
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.has_perm(
+            "algorithms.execute_algorithm", self.algorithm
+        ):
+            return self.handle_no_permission()
+        return super().dispatch(request, *args, **kwargs)
+
+
+class JobInterfaceSelect(
+    JobCreatePermissionMixin,
+    FormView,
+):
+    form_class = JobInterfaceSelectForm
+    template_name = "algorithms/job_form_create.html"
+    selected_interface = None
+
+    def get(self, request, *args, **kwargs):
+        if self.algorithm.interfaces.count() == 1:
+            self.selected_interface = self.algorithm.default_interface
+            return HttpResponseRedirect(self.get_success_url())
+        else:
+            return super().get(request, *args, **kwargs)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({"algorithm": self.algorithm})
+        return kwargs
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context.update(
+            {
+                "algorithm": self.algorithm,
+                "editors_job_limit": settings.ALGORITHM_IMAGES_COMPLIMENTARY_EDITOR_JOBS,
+            }
+        )
+        return context
+
+    def form_valid(self, form):
+        self.selected_interface = form.cleaned_data.pop("algorithm_interface")
```

Comment:
Is there a need to modify the data here rather than just accessing?

Reply:
Good point. Just accessing should be fine.